pytest/tests/loadtest/locust/common/social.py

```diff
@@ -0,0 +1,279 @@
+        super().__init__(contract_id, sender)
+        self.follow_list = follow_list
+
+    def build_args(self):
+        follow_list = self.follow_list
+        sender = self.sender.key.account_id
+        return social_follow_args(sender, follow_list)
+
+
+class InitSocialDB(Transaction):
+
+    def __init__(self, contract: Account):
+        super().__init__()
+        self.contract = contract
+
+    def sign_and_serialize(self, block_hash):
+        # Call the #[init] function, no arguments
+        call_new_action = create_function_call_action("new", "", 100 * TGAS, 0)
+
+        # Set the status to "Live" to enable normal usage
+        args = json.dumps({"status": "Live"}).encode('utf-8')
+        call_set_status_action = create_function_call_action(
+            "set_status", args, 100 * TGAS, 0)
+
+        # Batch the two actions above into one transaction
+        nonce = self.contract.use_nonce()
+        key = self.contract.key
+        return transaction.sign_and_serialize_transaction(
+            key.account_id, nonce, [call_new_action, call_set_status_action],
+            block_hash, key.account_id, key.decoded_pk(), key.decoded_sk())
+
+
+class InitSocialDbAccount(Transaction):
+    """
+    Send initial storage balance to ensure the account can use social DB.
+
+    Technically, we could also rely on lazy initialization and just send enough
+    balance with each request. But a typical user sends balance ahead of time.
+    """
+
+    def __init__(self, contract_id: str, account: Account):
+        super().__init__()
+        self.contract_id = contract_id
+        self.account = account
+
+    def sign_and_serialize(self, block_hash):
+        args = json.dumps({"account_id": self.account.key.account_id})
+        return transaction.sign_function_call_tx(self.account.key,
+                                                 self.contract_id,
+                                                 "storage_deposit",
+                                                 args.encode('utf-8'),
+                                                 300 * TGAS, 1 * NEAR_BASE,
+                                                 self.account.use_nonce(),
+                                                 block_hash)
+
+
+def social_db_build_index_obj(key_list_pairs: dict) -> dict:
+    """
+    JSON serializes the key - value-list pairs to be included in a SocialDB set message.
+
+    To elaborate a bit more, SocialDB expects for example
+    ```json
+    "index": { "graph": value_string } 
+    ```
+    where value_string = 
+    ```json
+    "[{\"key\":\"follow\",\"value\":{\"type\":\"follow\",\"accountId\":\"pagodaplatform.near\"}}]"
+    ```
+    So it's really JSON nested inside a JSON string.
+    And worse, the nested JSON is always a list of objects with "key" and "value" fields.
+    This method unfolds this format from a leaner definition, using a dict to
+    define each `value_string`.
+    """
+    obj = {}
```

Comment:
One way to write this in a more functional style would be to define a helper function:

```python

def social_db_build_index_obj(key_list_pairs: dict) -> dict:
  def serialize_values(values):
    return json.dumps([{"key": k, "value": v} for k, v in values.items()])
  
  return { key: serialize_values(values) for key, values in key_list_pairs.items() }
```

Reply:
I like that! I used your suggestion and then updated it according to the changes I had to do to fix another issue